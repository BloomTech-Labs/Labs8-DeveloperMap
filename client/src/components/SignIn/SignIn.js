import React from 'react';
import { ModalContainer, ModalMain } from '../../styles/ModalGlobalStyle.js';
import {
  Label,
  Input,
  SignModalMain,
  RegisterButton,
} from '../../styles/SignIn_UpStyle';

class SignIn extends React.Component {
  //Basic state
  state = {
    email: '',
    password: '',
  };

  changeHandler = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  clickRegister = () => {
    this.props.history.push('/Signup');
  };

  render() {
    console.log('%cstate', 'color: blue', this.state);
    return (
      <ModalContainer data-type="map">
        <SignModalMain>
          <h2>{this.props.currentSignedInUser ? `Welcome ${this.props.currentSignedInUser.email}` : 'Sign In'}</h2>
          <form onSubmit={(e) => {
            this.props.signInWithEmailAndPassword(e, this.state.email, this.state.password)
            this.setState({email: '', password: ''});
            }}>
            <Label htmlFor="email">
              Email
              <Input
                name="email"
                type="email"
                value={this.state.email}
                onChange={this.changeHandler}
              />
            </Label>
            <Label htmlFor="password">
              Password
              <Input
                name="password"
                type="password"
                value={this.state.password}
                onChange={this.changeHandler}
              />
            </Label>
            <button>Sign In</button>
            <p>
              New here?{' '}
              <RegisterButton onClick={this.clickRegister}>
                Register here
              </RegisterButton>
            </p>
          </form>
          <button onClick = {(e) => {
            this.props.signOutCurrentUser(e)
            }}>Sign Out</button>
        </SignModalMain>
      </ModalContainer>
    );
  }
}

export default SignIn;
