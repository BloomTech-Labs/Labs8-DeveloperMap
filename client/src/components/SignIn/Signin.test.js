import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from './SignIn';

jest.mock('../../firebase/firebase', () => {
  const firebasemock = require('firebase-mock');

  const mockdatabase = new firebasemock.MockFirebase();
  const mockauth = new firebasemock.MockFirebase();
  const mocksdk = new firebasemock.MockFirebaseSdk(
    path => {
      return path ? mockdatabase.child(path) : mockdatabase;
    },
    () => {
      return mockauth;
    }
  );

  const firebase = mocksdk.initializeApp(); // can take a path arg to database url
  // optional - expose the mock
  global.firebase = firebase;

  // return the mock to match your export api
  return firebase;
});

describe(`seeker profile component`, () => {
  it(`should render without error`, () => {
    const props = {
      match: {
        params: {
          seekerId: 'uid1',
        },
      },
    };

    const tree = renderer.create(<SignIn {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});