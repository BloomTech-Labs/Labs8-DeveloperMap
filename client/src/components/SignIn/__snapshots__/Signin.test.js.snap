// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seeker profile component should render without error 1`] = `
<div
  className="sc-bdVaJa cbxIaT"
  data-type="modal-container"
>
  <div
    className="sc-bwzfXH modal sc-htpNat FdupV"
    width="21%"
  >
    <h2>
      Sign In
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-htoDjs gaXcgg"
      >
        <input
          className="sc-ifAKCX hEAxUp"
          id="email"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
        <label
          className="sc-bxivhb jhvOIQ"
          htmlFor="email"
        >
          Email
        </label>
      </div>
      <div
        className="sc-htoDjs gaXcgg"
      >
        <input
          className="sc-ifAKCX hEAxUp"
          id="password"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
        <label
          className="sc-bxivhb jhvOIQ"
          htmlFor="password"
        >
          Password
        </label>
      </div>
      <button
        className="sc-gqjmRU kVfWaB"
      >
        Sign In
      </button>
      - or -
      <button
        className="sc-gqjmRU kVfWaB"
        onClick={[Function]}
      >
        Register
      </button>
      <div
        className="sc-VigVT fFMZAF"
        onClick={[Function]}
      />
      <div
        className="sc-jTzLTM euLUTA"
        onClick={[Function]}
      />
    </form>
  </div>
</div>
`;
